User-Data Store Migration/dump-converter.py

```diff
@@ -138,26 +139,32 @@ def writeline(stream, line):
 def initialize(out):
     global pre, replace_rules, post, NAME_TEMPLATE
 
+    # Delete any action on the internal sqlite_sequence table
     replace_rules.append((True, r"^.*sqlite_sequence.*;[\r\n]*$", "", 0, None, True))
-    replace_rules.append((True, r"^PRAGMA foreign_keys=OFF;[\r\n]*$", "", 0, None, True))
+    # Fix the datetime values
     replace_rules.append((False, " +00:00'", "'", 0, None, False))
-    replace_rules.append((True, r"^INSERT INTO.*$", None, 0, injectStructureInfoInsertInto, False))
+    # Rewrite compatible insert rules
+    replace_rules.append((True, r"^INSERT INTO.*$", None, 0, injectStructureInfoInsertInto, True))
 
     if __dialect__ in ["mysql", "mariadb"]:
         NAME_TEMPLATE = "`%s`"
+        pre.append("START TRANSACTION;")
+        # Disable/enable the constraint checks
         pre.append("SET FOREIGN_KEY_CHECKS=0;")
         post.append("SET FOREIGN_KEY_CHECKS=1;")
         
-        replace_rules.append((True, r"^BEGIN TRANSACTION;\s*$", "START TRANSACTION;", 0, None, True))

```

Comment:
I also removed the transformation rules for the transactions handling by explicitly adding the queries with the right syntax